### PR TITLE
[Snyk] Security upgrade pydantic from 2.3.0 to 2.4.0

### PR DESCRIPTION
**Pull request by m31-galaxy**
<p>This PR was automatically created by Snyk using the credentials of a real user.</p><br /><h3>Snyk has created this PR to fix one or more vulnerable packages in the `pip` dependencies of this project.</h3>

#### Changes included in this PR

- Changes to the following files to upgrade the vulnerable dependencies to a fixed version:
    - requirements.txt

##### By pinning:
Severity                   | Priority Score (*)                   | Issue                   | Upgrade                   | Breaking Change                   | Exploit Maturity
:-------------------------:|-------------------------|:-------------------------|:-------------------------|:-------------------------|:-------------------------
medium severity  |  **658/1000**  <br/> **Why?** Proof of Concept exploit, Recently disclosed, Has a fix available, CVSS 5.3  | Regular Expression Denial of Service (ReDoS) <br/>[SNYK-PYTHON-PYDANTIC-5907722](https://snyk.io/vuln/SNYK-PYTHON-PYDANTIC-5907722) |  `pydantic:` <br> `2.3.0 -> 2.4.0` <br>  |  No  | Proof of Concept 

(*) Note that the real score may have changed since the PR was raised.

Some vulnerabilities couldn't be fully fixed and so Snyk will still find them when the project is tested again. This may be because the vulnerability existed within more than one direct dependency, but not all of the affected dependencies could be upgraded.

Check the changes in this PR to ensure they won't cause issues with your project.

------------

**Note:** *You are seeing this because you or someone else with access to this repository has authorized Snyk to open fix PRs.*

For more information:  
🧐 [View latest project report](https://app.snyk.io/org/thatotherandrew-4i6/project/72587030-c41a-4c1e-ad3d-277b39bf330e?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;fix-pr)

🛠 [Adjust project settings](https://app.snyk.io/org/thatotherandrew-4i6/project/72587030-c41a-4c1e-ad3d-277b39bf330e?utm_source&#x3D;github&amp;utm_medium&#x3D;referral&amp;page&#x3D;fix-pr/settings)

📚 [Read more about Snyk's upgrade and patch logic](https://support.snyk.io/hc/en-us/articles/360003891078-Snyk-patches-to-fix-vulnerabilities)

[//]: # (snyk:metadata:{"prId":"deeaa15d-df22-4320-89aa-326a6f8335a5","prPublicId":"deeaa15d-df22-4320-89aa-326a6f8335a5","dependencies":[{"name":"pydantic","from":"2.3.0","to":"2.4.0"}],"packageManager":"pip","projectPublicId":"72587030-c41a-4c1e-ad3d-277b39bf330e","projectUrl":"https://app.snyk.io/org/thatotherandrew-4i6/project/72587030-c41a-4c1e-ad3d-277b39bf330e?utm_source=github&utm_medium=referral&page=fix-pr","type":"auto","patch":[],"vulns":["SNYK-PYTHON-PYDANTIC-5907722"],"upgrade":[],"isBreakingChange":false,"env":"prod","prType":"fix","templateVariants":["updated-fix-title","priorityScore"],"priorityScoreList":[658],"remediationStrategy":"vuln"})

---

**Learn how to fix vulnerabilities with free interactive lessons:**

 🦉 [Regular Expression Denial of Service (ReDoS)](https://learn.snyk.io/lesson/redos/?loc&#x3D;fix-pr)

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -2,7 +2,7 @@ aiofiles==23.2.1
 aiohttp==3.8.5
 discord.py==2.3.2
 packaging==23.1
-pydantic==2.3.0
+pydantic==2.4.0
```
Comment (Fripe070) [inline requirements.txt]:
```suggestion
pydantic==2.4.2
```
[pydantic 2.4.2](https://github.com/pydantic/pydantic/releases/tag/v2.4.2) has been released, if an update is to be done it should probably be to that.

**Review by Fripe070 — CHANGES_REQUESTED** [review]
Update pydantic directly to 2.4.1

File: requirements.txt (hunk 1 above)
Comment (m31-galaxy) [reply requirements.txt]: So is 2.4.2, maybe we should upgrade directly to that instead?

Comment (Fripe070) [reply requirements.txt]: I didn't even notice it was out as it isn't on their [changelog page](https://docs.pydantic.dev/latest/changelog/), but yeah that'd be better.
The review has been updated to reflect that

**Review by Fripe070 — APPROVED** [review]

**Pull request merged**